lint-workflow/lint.py

```diff
@@ -3,6 +3,116 @@
 import yaml
 import json
 import urllib3 as urllib
+import sys
+from urllib3.util import Retry
+from urllib3.exceptions import MaxRetryError
+import logging
+
+PROBLEM_LEVELS = {
+    "warning": 1,
+    "error": 2,
+}
+
+
+class Colors:
+    """Class containing color codes for printing strings to output."""
+
+    black = "30m"
+    red = "31m"
+    green = "32m"
+    yellow = "33m"
+    blue = "34m"
+    magenta = "35m"
+    cyan = "36m"
+    white = "37m"
+
+
+class LintFinding(object):
+    """Represents a linting problem."""
+
+    def __init__(self, description="<no description>", level=None):
+        self.description = description
+        self.level = level
+
+
+def get_max_error_level(findings):
+    """Get max error level from list of findings."""
+    max_problem= max(findings, key=lambda finding: PROBLEM_LEVELS[finding.level])
+    max_problem_level=PROBLEM_LEVELS[max_problem.level]
+    return max_problem_level
+
+
+def print_finding(finding: LintFinding):
+    """Print formatted and colored finding."""
+    if finding.level == "warning":
+        color = Colors.yellow
+    elif finding.level == "error":
+        color = Colors.red
+    else:
+        color = Colors.white
+
+    line = f"  - \033[{color}{finding.level}\033[0m {finding.description}"
+
+    print(line)
+
+
+def get_github_api_response(url, action_id):
+    """Call GitHub API with error logging without throwing an exception."""
+    http = urllib.PoolManager()
+    headers = {"user-agent": "bw-linter"}
+
+    if os.getenv("GITHUB_TOKEN", None):
+        headers["Authorization"] = f"Token {os.environ['GITHUB_TOKEN']}"
+
+    message = f"Failed to call GitHub API for action: {action_id} due to "
+
+    try:
+        response = http.request("GET", url, headers=headers)
+    except:
+        e = sys.exc_info()[0]
+        message += e
+        logging.error(message)
+        return None
+
+    if response.status == 403 and response.reason == "rate limit exceeded":
+        logging.error(f"Failed to call GitHub API for action: {action_id} due to rate limit exceeded.")
+        # Handle github api limit exceed by returning that the action exists without actually checking
+        # to prevent false errors on linter output. Only show it as an linter error.
```

Comment:
I lean towards leaving the linter code errors, and not display them in the problems with the file itself. I think that way people will not get confused about the warnings which are actually not the action yml file's fault.